src/main/java/com/direwolf20/buildinggadgets/common/tools/ExchangingModes.java

```diff
@@ -16,26 +15,154 @@
+            }
+        }
+
+        return coordinates;
+    }),
+    VerticalColumn((world, player, hit, sideHit, tool) -> {
+        List<BlockPos> coordinates = new ArrayList<>();
+        IBlockState target = GadgetUtils.getToolActualBlock(tool);
+        IBlockState current = world.getBlockState(hit);
+        int range = GadgetUtils.getToolRange(tool);
+        int radius = (range - 1) / 2;
+        boolean fuzzyMode = GadgetGeneric.getFuzzy(tool);
+
+        BlockPos base;
+        EnumFacing extension;
+        if (sideHit.getAxis().isVertical()) {
+            EnumFacing playerFacing = player.getHorizontalFacing();
+            base = hit.offset(playerFacing.getOpposite(), radius);
+            extension = playerFacing;
+        } else {
+            base = hit.down(radius);
+            extension = EnumFacing.UP;
+        }
+
+        for (int i = 0; i < range; i++) {
+            BlockPos pos = base.offset(extension, i);
+            if (isReplaceable(world, pos, current, target, fuzzyMode)) {
+                coordinates.add(pos);
+            }
+        }
+
+        return coordinates;
+    }),
+    HorizontalColumn((world, player, hit, sideHit, tool) -> {
+        List<BlockPos> coordinates = new ArrayList<>();
+        IBlockState target = GadgetUtils.getToolActualBlock(tool);
+        IBlockState current = world.getBlockState(hit);
+        int range = GadgetUtils.getToolRange(tool);
+        int radius = (range - 1) / 2;
+        boolean fuzzyMode = GadgetGeneric.getFuzzy(tool);
+
+        EnumFacing playerFacing = player.getHorizontalFacing();
+        EnumFacing extension = playerFacing.rotateY();
+        BlockPos base = hit.offset(extension.getOpposite(), radius);
+
+        for (int i = 0; i < range; i++) {
+            BlockPos pos = base.offset(extension, i);
+            if (isReplaceable(world, pos, current, target, fuzzyMode)) {
+                coordinates.add(pos);
+            }
+        }
+
+        return coordinates;
+    }),
+    Grid((world, player, hit, sideHit, tool) -> {
+        List<BlockPos> coordinates = new ArrayList<>();
+        IBlockState target = GadgetUtils.getToolActualBlock(tool);
+        IBlockState current = world.getBlockState(hit);
+        int range = GadgetUtils.getToolRange(tool) + 1;
+        boolean fuzzyMode = GadgetGeneric.getFuzzy(tool);
+
+        for (int x = range * -7 / 5; x <= range * 7 / 5; x++) {
+            for (int z = range * -7 / 5; z <= range * 7 / 5; z++) {
+                if (x % (((range - 2) % 6) + 2) == 0 && z % (((range - 2) % 6) + 2) == 0) {
+                    BlockPos pos = new BlockPos(hit.getX() + x, hit.getY(), hit.getZ() + z);
+                    if (isReplaceable(world, pos, current, target, fuzzyMode)) {
+                        coordinates.add(pos);
+                    }
+                }
+            }
+        }
+
+        return coordinates;
+    });
+
+    private final String displayName;
+    private final IBuildingMode modeImpl;
```

Comment:
could this be moved to an abstract Method and then IBuildingMode expanded beyond the simple lambda method, and instead provide some kind of registration facility, so that other people can register (and maybe unregister?) modes. Maybe using a RecourceLocation as keys for the modes?

Reply:
Rewriting.